### PR TITLE
append index name for the source of the cluster put-mapping task

### PR DESCRIPTION
**Pull request by zhanghezhen**
When there are a lot of pending tasks with put-mapping in the cluster, the "_cat/pending_tasks" API result is too simple to quickly know which index is put-mapping.
Since the task desc is obtained from the 'source' field of task, we can append the index name after the source field when pending task is submitted, it is helpful for us to quickly locate the put-mapping index

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataMappingService.java [hunk 1]
```diff
@@ -326,7 +326,7 @@ private ClusterState applyRequest(ClusterState currentState, PutMappingClusterSt
     }
 
     public void putMapping(final PutMappingClusterStateUpdateRequest request, final ActionListener<ClusterStateUpdateResponse> listener) {
-        clusterService.submitStateUpdateTask("put-mapping",
+        clusterService.submitStateUpdateTask("put-mapping " + getRequestIndex(request),
```
Comment (henningandersen) [inline server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataMappingService.java]:
```suggestion
        clusterService.submitStateUpdateTask("put-mapping " + Strings.arrayToCommaDelimitedString(request.indices())),
```

This removes the need for the method below too.

File: server/src/test/java/org/elasticsearch/cluster/metadata/MetaDataMappingServiceTests.java [hunk 2]
```diff
@@ -113,4 +113,17 @@ public void testMappingVersionUnchanged() throws Exception {
         assertThat(result.resultingState.metaData().index("test").getMappingVersion(), equalTo(previousVersion));
     }
 
+    public void testGetRequestIndex() {
```
Comment (henningandersen) [inline server/src/test/java/org/elasticsearch/cluster/metadata/MetaDataMappingServiceTests.java]: Given that we have the comma concatenating logic, I think we can omit the test here.

**Review by henningandersen — CHANGES_REQUESTED** [review]
@zhanghezhen thanks for your PR. It looks like we can simplify it a bit, see comments.

File: server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataMappingService.java (hunk 1 above)
Comment (zhanghezhen) [reply server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataMappingService.java]: that's great

File: server/src/test/java/org/elasticsearch/cluster/metadata/MetaDataMappingServiceTests.java (hunk 2 above)
Comment (zhanghezhen) [reply server/src/test/java/org/elasticsearch/cluster/metadata/MetaDataMappingServiceTests.java]: Of course

**Comment by henningandersen** [comment]
@elasticmachine test this please

**Comment by henningandersen** [comment]
@elasticmachine update branch

**Comment by henningandersen** [comment]
@elasticmachine test this please

**Review by henningandersen — COMMENTED** [review]
LGTM, thanks @zhanghezhen

**Review by henningandersen — APPROVED** [review]

**Pull request merged**